src/checkers.cpp

```diff
@@ -7,34 +7,55 @@ namespace ada::checkers {
 
   // TODO: Refactor this to not use `std::vector` but use pointer arithmetic for performance.
   bool ends_in_a_number(const std::string_view input) noexcept {
-    // Let parts be the result of strictly splitting input on U+002E (.).
-    std::vector<std::string> parts = ada::helpers::split_string_view(input, '.', false);
```

Comment:
Getting rid of std::vector in this code is assuredly a good step forward. Please merge.

std::vector will tend to allocate on the heap. You want to avoid that as much as possible. It is expensive in our context.